agent/consul/gateways/controller_gateways.go

```diff
@@ -248,6 +248,16 @@ func (r *apiGatewayReconciler) reconcileGateway(_ context.Context, req controlle
 	if len(certificateErrors) > 0 {
 		updater.SetCondition(invalidCertificates())
 	} else {
+		meta.eachListener(func(listener *structs.APIGatewayListener, bound *structs.BoundAPIGatewayListener) error {
+			listenerRef := structs.ResourceReference{
+				Kind:           structs.APIGateway,
+				Name:           meta.BoundGateway.Name,
+				SectionName:    bound.Name,
+				EnterpriseMeta: meta.BoundGateway.EnterpriseMeta,
+			}
+			updater.SetCondition(validCertificate(listenerRef))
+			return nil
+		})
```

Comment:
So, I think what you'd actually need to do here is move this call above the iteration on line 244. Take for instance if one listener has valid certificate refs and the other does not, what will happen?

1. The first, valid listener, will get no `ResolvedRefs` status condition
2. The second, invalid listener, will get `ResolvedRefs == False`

What we want is for:

1. The first, valid listener, to get `ResolvedRefs == True`
2. The second, invalid listener, to get `ResolvedRefs == False`

Since calling `SetCondition` will override the condition with the same `Type` and `Ref`, then we should be able to just set both of them to `ResolvedRefs == True` initially and use the range in line 244 to override them to `ResolvedRefs == False` when we encounter an error. You should be able to add a test that verifies the whole matrix of:

1. all listeners valid refs
2. all listeners invalid refs
3. mixed valid/invalid refs

checking the `ResolvedRefs` and `Accepted` statuses (with `InvalidCertificates`) of the gateway after.

Reply:
ahhh gotcha, that makes sense, lemme update this